collections/bitroomgrid.rs

```diff
@@ -0,0 +1,248 @@
+//! Dependencies: `serde`
+//!
+//! A 50×50 binary grid meant to be used to flag tiles in a room
+//!
+//! Has basic compression built in too for more compact serialization
+
+
+use serde::de::{self, Deserialize, Deserializer, Visitor};
+use serde::ser::{self, Serialize, Serializer};
+use std::fmt::{self, Write};
+
+const ROWS: usize = 10;
+const ROW_SIZE: usize = 32 * 8; // Each row holds 32 * 8 = 256 bits of data
+
+/// 50×50 binary flag map
+#[derive(Debug, Clone)]
+pub struct BitRoomGrid {
+    buffer: [[u8; ROW_SIZE / 8]; ROWS],
```

Comment:
I think you mentioned why already on slack, but why not use 
https://docs.rs/bitvec/0.14.0/bitvec/ ?

Reply:
Yet, why require it uses a 2d grid, rather than a flat vector? I feel like it'd be easier to implement if it were flat?